helm/aws-load-balancer-controller/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,30 @@
+{{- if.Values.serviceMonitor.enabled -}}
+apiVersion: monitoring.coreos.com/v1
```

Comment:
I agree about the documentation, but testing the API server often doesn't work as expected (`helm template` and ArgoCD come to mind) so I'd advise against this approach.

Reply:
It may not work with ArgoCD, but IMO still valuable for helm install/upgrade workflows. The `helm template` with `--validate` flag also honors the api server capabilities. For workflows that doesn't support API capabilities check, doesn't work either way, so why not include the check?

Reply:
@kishorj every time I've added complex checks into charts the ArgoCD users (or users who don't know to use `helm template --validate`) have pushed back. I'm happy to add the check as it'd work for my use case, but I just want to make sure that we've considered this as a downside. Personally I'd still leave it without the check as it's an opt-in setting that will either work or error so the cost seems to outweigh the benefit but it's your call.

Reply:
if the CRD is not installed, but user enables serviceMonitor, does the chart installation fail or we'd end up with a partial installation?

Reply:
I will check, but the behaviour as coded would be consistent with the other use of CRDs for cert-manager which only checks for v1 not if the CRD type exists at all.